compiler/modules/front.syntax/parseExpr.bal

```diff
@@ -448,8 +451,9 @@ function parseSimpleConstExpr(Tokenizer tok) returns SimpleConstExpr|err:Syntax
     }
     match t {
         [IDENTIFIER, var identifier] => {
+            Position identifierPos = tok.currentStartPos();
             check tok.advance();
-            var [prefix, varName] = check parseOptQualIdentifier(tok, identifier);
+            var [prefix, varName, _] = check parseOptQualIdentifier(tok, identifier, identifierPos);
```

Comment:
Doesn't look right to ignore the position. Shouldn't we be using the position of the identifier below in the constructor instead of startPos?

Reply:
this is parsing a match pattern of the form `a[:b]` startPos the the startPosition of a and the position ignored will the start position of b (if we have a "b" else it will the start position of "a")